src/api/forms/FormsAPI.js

```diff
@@ -158,10 +162,12 @@ export default class FormsAPI extends EventEmitter {
                     key = property.join('.');
                 }
 
-                changes[key] = data.value;
+                _.set(changes, `${key}`, data.value);
+                self.emit('formPropertiesChanged', changes);
```

Comment:
Why are we emitting an event here _and_ calling the `onchange` callback. It's going to be very confusing to developers why we have both event mechanisms.